handout/testing/unit_testing_redux.md

```diff
@@ -0,0 +1,229 @@
+# Unit Testing Redux
+
+The units that make up redux are actions and reducers. Lets try testing them.
+
+## Testing Synchronous Actions
+
+Recall that by synchronous we mean action creators that return a simple JSON object, like the first two counter actions in [ng2-redux-starter](https://github.com/rangle/angular2-redux-starter).   
+
+```ts
+export const INCREMENT_COUNTER = 'INCREMENT_COUNTER';
+export const DECREMENT_COUNTER = 'DECREMENT_COUNTER';
+
+export function increment() {
+  return {
+    type: INCREMENT_COUNTER
+  };
+}
+
+export function decrement() {
+  return {
+    type: DECREMENT_COUNTER
+  };
+}
+
+...
+
+```
+
+These are pretty straightforward to test.
+
+```ts
+import {INCREMENT_COUNTER, DECREMENT_COUNTER} from './counter';
+import * as CounterActions from './counter';
+
+describe('counter action creators', () => {                     
+  it('increment should create INCREMENT_COUNTER action', () => {
+    chai.expect(CounterActions.increment())                     
+      .to.deep.equal({                                          
+        type: INCREMENT_COUNTER                                 
+      });                                                       
+  });                                                           
+
+  it('decrement should create DECREMENT_COUNTER action', () => {
+    chai.expect(CounterActions.decrement())                     
+      .to.deep.equal({                                          
+        type: DECREMENT_COUNTER                                 
+      });                                                       
+  });                                                           
+
+  ...
+
+});                                                             
+```
+
+We just make sure that our action creators do indeed create actions.
```

Comment:
We should probably make sure that actions are actions, as in that they are POJO's with a type property.